### ISSUE TITLE
Callback not called if no options are passed to publishOrQueue

### ISSUE THREAD

**Issue by scottgunther** [GannettDigital/busterbunny.js#5]
If options is null the length is never 5 for the callback to be set even if one is passed.
See commented code

``` javascript
function publishOrQueue(exchange, eventId, eventData, options, afterRaised) {
            var args = [exchange, eventId, eventData];
            if (options) args.push(options); // <<<<< Not pushed if falsey >>>>>
            if (afterRaised) args.push(afterRaised);
            _eventPublishQueue.push(args);

            if (_eventPublishQueue.length >= _thresholds.maxRaisedEvents) {
                self.emit(self.EVENTS.WARNING_RAISED, _eventPublishQueue.length + ' events queued is greater than or equal to max of ' + _thresholds.maxRaisedEvents);
            }

            self.emit(self.EVENTS.PUBLISH_REQUESTED);
        }

function publishQueuedRequestsRecursively() {
            if (!_connected || !_publishingChannel) return;

            var args = _eventPublishQueue.pop();

            if (!args) return;

            //the last arg is a method for after invoking publish
            var afterRaised;
            var error;

            if (args.length == 5) // <<<<< length is 4 even though callback provided >>>>>
                afterRaised = args.splice(args.length-1, 1)[0];
```

**Comment by jhaugh42** [comment]
Per our discussion this morning, we're going to reenforce the opinions of the framework in the following ways: 
1. eventId will be required to be typeof 'string'
2. afterRaised will be required to be typeof 'function'

This effectively requires all calls to the framework specify an eventId that is also a compatible routing key AND that all callers specify what to do when the framework is done (successfully or not) with operations.

@binaryalchemist: Can you verify that I've correctly enumerated these points from our discussion earlier? Thanks!

**Comment by jhaugh42** [comment]
Part one of this was resolved with #7
